res/controllers/Denon-MC7000-scripts.js

```diff
@@ -1036,26 +1216,85 @@ MC7000.HotCueLED = function(value, group) {
+                    midi.sendShortMsg(0x94 + deckIdx, 0x1C + samplerIdx - 1, MC7000.padColor.sampleroff); // shift pressed sets the same color
                 }
             }
         }
     }
 };
 
+// Velocity Sampler LED
+MC7000.VelSampLED = function() {
+    for (var deckIdx = 0; deckIdx < 4; deckIdx++) {
+        for (var velSampIdx = 1; velSampIdx <= 8; velSampIdx++) {
+            var sampNo = 0;
+            if (MC7000.PADModeVelSamp[deckIdx]) {
+                if (MC7000.SamplerQty === 16) {
+                    sampNo = (deckIdx % 2) * 8 + velSampIdx; // use sampler 1-16 for deck 1 and deck 2 and sampler 1-16 for deck 3 and 4
+                } else if (MC7000.SamplerQty === 32) {
+                    sampNo = deckIdx * 8 + velSampIdx; // use sampler 1 - sampler 32 for deck 1 - 4
+                }
+                if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 1) {
+                    var samplerIsNotPlaying = engine.getValue("[Sampler"+sampNo+"]", "play") === 0;
+                    if (samplerIsNotPlaying) {
+                        midi.sendShortMsg(0x94 + deckIdx, 0x14 + velSampIdx - 1, MC7000.padColor.velsamploaded);
+                        midi.sendShortMsg(0x94 + deckIdx, 0x1C + velSampIdx - 1, MC7000.padColor.velsamploaded); // shift pressed sets the same color
+                        engine.setValue("[Sampler"+sampNo+"]", "pregain", 1);
+                    } else {
+                        midi.sendShortMsg(0x94 + deckIdx, 0x14 + velSampIdx - 1, MC7000.padColor.velsampplay);
+                    }
+                } else if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 0) {
+                    midi.sendShortMsg(0x94 + deckIdx, 0x14 + velSampIdx - 1, MC7000.padColor.velsampoff);
+                    midi.sendShortMsg(0x94 + deckIdx, 0x1C + velSampIdx - 1, MC7000.padColor.velsampoff); // shift pressed sets the same color
+                }
+            }
+        }
+    }
+};
+
+// Pitch LED when loading a track with already existing hotcues for pitch
+MC7000.PitchLED = function(value, group) {
+    var deckNumber = script.deckFromGroup(group);
+    var deckIndex = deckNumber - 1;
+    if (MC7000.PADModePitch[deckIndex]) {
+        for (var pitchIdx = 0; pitchIdx < 8; pitchIdx++) {
+            if (engine.getValue("[Channel"+deckNumber+"]", "play") === 0) { // stopped
+                if (MC7000.HotcueSelectedGroup[deckIndex] !== 0) { // hotcue selected
+                    midi.sendShortMsg(0x94 + deckIndex, 0x14 + pitchIdx, MC7000.padColor.pitchoff);
+                } else {
+                    var hotcueEnabled = engine.getValue(group, "hotcue_" + pitchIdx + 1 + "_enabled", true);
+                    midi.sendShortMsg(0x94 + deckIndex, 0x14 + pitchIdx, hotcueEnabled ? MC7000.padColor.hotcueon : MC7000.padColor.hotcueoff);
+                }
+            } else {
+                midi.sendShortMsg(0x94 + deckIndex, 0x14 + pitchIdx, MC7000.padColor.pitchon);
```

Comment:
Why is it enough to send the color only once in this case?

Reply:
This function is not executed in relation to button press, so also no shift button here.
Please see around line 688: `midi.sendShortMsg(0x94 + deckIndex, 0x1C + pitchIdx - 1, MC7000.padColor.pitchon); // keep color when shift is pressed`